src/views/components/FeedListView/index.js

```diff
@@ -1,16 +1,51 @@
-import React from 'react';
+import React, {useState} from 'react';
 import PropTypes from 'prop-types';
+import Button from '../Button';
 
 function FeedListView(props) {
-  return (
+  const [state, setState] = useState({readingId: null});
+
+  const toggleItem = (id) => {
+    setState((prevState) => ({
+      ...prevState,
+      readingId: id
+    }));
+  };
```

Comment:
Does the previous state need to be saved? the state can be simplified to just be the `readingId` of the message, instead of it being an object.

Reply:
I had that thought too but I followed @PhillyWebGuy's convention and it actually occurred to me that it might be kind of helpful if we wanted to add more things to state. The question though is how idiomatic that is.

Reply:
in react, it is conventional to potentially have multiple `useState` hooks in one component, each for their purpose.

You don't want to jam every possible piece of state into one useState hook because that might actually cause unintentional re-renders or state changes to occur across multiple DOM elements.

Reply:
previously, when components were designed using classes with `React.Component`, pretty much all state variables went into one large blob

Since the introduction of hooks, things have shifted away from that for good reason. If you have a react component where you have multiple `useStates`, it is an indication that you need to break that component up into smaller functional components.

Reply:
That tracks with what I figured. Here's my suggestion - this is working so let's add it to our list of "things we _will_ improve on".

Reply:
I approved the PR. This is one of those "I am going to mention this so that it gets put on the table, whether it gets addressed now or sometime later when we come back to make improvements."